### DISCUSSION TITLE
Is this an issue with Application Gateway or AGIC updating it?

### DISCUSSION THREAD

**Discussion by robs** [Azure/application-gateway-kubernetes-ingress#1157] [Q&A]
Hiya,

We had AGIC setup and working fine until we deployed the "production" app which then was intercepting ALL traffic.

These routes were working fine together:
- api.dev.site.com/xxx-api/* = dev-xxx-api
- api.beta.site.com/xxx-api/* = beta-xxx-api
- api.site.com/xxx-api/* = prod-xxx-api
- \*.dev.site.com/* = dev-site
- \*.beta.site.com/* = beta-site

When we deployed the 'prod' site:
- \*.site.com/* = prod-site

Everything was directed to that!
It's like app gateway was using the LEAST specific rule rather than MOST specific.

All of the listeners are multi-site, well, they all have hostname:null and hostnames populated.

Is that an issue with app gateway or the way AGIC updates it?

I can't believe it's how app gateway is supposed to behave with this config.

Thanks,

Rob.

**Comment by robs** [comment]
Looks like App Gateway just uses the first rule that matches so whenever it hits \*.site.com/* all other rules are ignored.

There is a "priority" property on the routing rules but I can't see any annotations that look like they'd set that.

Anyone know how to set the priority of the rules?

**Comment by akshaysngupta** [comment]
Hi @robs, yes, that's correct. Here is [an article ](https://docs.microsoft.com/en-us/azure/application-gateway/configuration-request-routing-rules#order-of-processing-rules)explaining the behavior.

**Reply by robs** [reply]
I raised an issue [here](https://github.com/Azure/application-gateway-kubernetes-ingress/issues/1159) to add an annotation that allows setting the priority of the rules.

If I could set the priority through AGIC that would resolve the issue wouldn't it?
